x/liquidstakeibc/keeper/icq.go

```diff
@@ -188,3 +192,64 @@ func RewardsAccountBalanceCallback(k Keeper, ctx sdk.Context, data []byte, query
 
 	return nil
 }
+
+func NonCompoundableRewardsAccountBalanceCallback(k Keeper, ctx sdk.Context, data []byte, query icqtypes.Query) error {
+	hc, found := k.GetHostChain(ctx, query.ChainId)
+	if !found {
+		return fmt.Errorf("host chain with id %s is not registered", query.ChainId)
+	}
+
+	balance, err := bankkeeper.UnmarshalBalanceCompat(k.cdc, data, hc.RewardParams.Denom)
+	if err != nil {
+		return fmt.Errorf("could unmarshal balance from ICQ balances request: %w", err)
+	}
+
+	hc.RewardsAccount.Balance = balance
+	if !hc.RewardsAccount.Balance.IsZero() {
+
+		// limit the auto-compounded rewards to the host chain autocompound factor
+		var autocompoundRewards sdk.Coin
+		maxAmountToTransfer := sdk.NewDecFromInt(hc.GetHostChainTotalDelegations()).Mul(hc.AutoCompoundFactor).TruncateInt()
+		if maxAmountToTransfer.GT(hc.RewardsAccount.Balance.Amount) {
+			autocompoundRewards = hc.RewardsAccount.Balance
+		} else {
+			autocompoundRewards = sdk.NewCoin(hc.RewardsAccount.Balance.Denom, maxAmountToTransfer)
+		}
+
+		// build the transfer message to send the rewards to the swapping address
+		msgTransfer := &banktypes.MsgSend{
+			FromAddress: hc.RewardsAccount.Address,
+			ToAddress:   hc.RewardParams.Destination,
+			Amount:      sdk.NewCoins(autocompoundRewards),
+		}
+
+		// execute the ICA transfer transaction
+		_, err = k.GenerateAndExecuteICATx(
```

Comment:
is the ica Ack/ Timeout handled?

Reply:
if it times out, will retry the next epoch

Reply:
i see, and how does dydx compounding happen?

Reply:
script will take rewards from the destination address, swap them, and send them to the deposit address directly

Reply:
deposit address on persistence or host_chain? 
Is there deposit (delegatable or just deposit) entry being added?

Reply:
the flow will be the following:

1. Module queries for the balance of the denom inside RewardParams.
2. Upon response, the max compoundable amount is sent to the destination address inside RewardParams.
3. Job will swap rewards on Dexter (routing them properly).
4. Rewards are sent back to the reward ICA account.
5. Module picks up rewards as normally.